code/game/objects/structures/platforms.dm

```diff
@@ -1,156 +1,175 @@
 
-/obj/structure/platform/proc/on_try_exit(datum/source, atom/movable/O, direction, list/knownblockers)
+	var/static/list/entry_only = list(COMSIG_ATOM_ENTERED = PROC_REF(on_enter))
+
+	var/static/list/exit_only = list(COMSIG_ATOM_EXITED = PROC_REF(on_exit))
+
+
+	AddElement(/datum/element/connect_loc, (platform_flags & ENTRY_SLOWDOWN) ? (platform_flags & EXIT_SLOWDOWN) ? entry_and_exit : entry_only : exit_only)
+
+///clears and sets overlays based on current dir
+/obj/structure/platform/proc/apply_overlays()
+	overlays.Cut()
+	var/image/new_overlay
+
+	if(dir & EAST)
+		new_overlay = image(icon, src, "[initial(icon_state)]_overlay", layer, EAST)
+		new_overlay.pixel_x = 32
+		overlays += new_overlay
+
+	if(dir & WEST)
+		new_overlay = image(icon, src, "[initial(icon_state)]_overlay", layer, WEST)
+		new_overlay.pixel_x = -32
+		overlays += new_overlay
+
+	if(dir & NORTH)
+		new_overlay = image(icon, src, "[initial(icon_state)]_overlay", layer, NORTH)
+		new_overlay.pixel_y = 32
+		new_overlay.layer = ABOVE_MOB_LAYER //perspective
+		overlays += new_overlay
+
+	if(dir & SOUTH)
+		new_overlay = image(icon, src, "[initial(icon_state)]_overlay", layer, SOUTH)
+		new_overlay.pixel_y = -32
+		overlays += new_overlay
+
+	if(CHECK_MULTIPLE_BITFIELDS(dir, NORTHEAST))
+		new_overlay = image(icon, src, "[initial(icon_state)]_overlay", layer, NORTHEAST)
+		new_overlay.pixel_y = 32
+		new_overlay.pixel_x = 32
+		new_overlay.layer = ABOVE_MOB_PLATFORM_LAYER
+		overlays += new_overlay
+
+	if(CHECK_MULTIPLE_BITFIELDS(dir, NORTHWEST))
+		new_overlay = image(icon, src, "[initial(icon_state)]_overlay", layer, NORTHWEST)
+		new_overlay.pixel_y = 32
+		new_overlay.pixel_x = -32
+		new_overlay.layer = ABOVE_MOB_PLATFORM_LAYER
+		overlays += new_overlay
+
+	if(CHECK_MULTIPLE_BITFIELDS(dir, SOUTHEAST))
+		new_overlay = image(icon, src, "[initial(icon_state)]_overlay", layer, SOUTHEAST)
+		new_overlay.pixel_y = -32
+		new_overlay.pixel_x = 32
+		overlays += new_overlay
+
+	if(CHECK_MULTIPLE_BITFIELDS(dir, SOUTHWEST))
+		new_overlay = image(icon, src, "[initial(icon_state)]_overlay", layer, SOUTHWEST)
+		new_overlay.pixel_y = -32
+		new_overlay.pixel_x = -32
+		overlays += new_overlay
+
+///Applies slowdown when entering if applicable
+/obj/structure/platform/proc/on_enter(datum/source, atom/movable/arrived, atom/old_loc, list/atom/old_locs)
 	SIGNAL_HANDLER
-	if(O.throwing)
+	if(!ismob(arrived))
+		return
+	if(arrived.throwing)
 		return NONE
-	if(!density || !(flags_atom & ON_BORDER) || !(direction & dir) || (O.status_flags & INCORPOREAL))
+	if((arrived.status_flags & INCORPOREAL))
+		return
+	if(!isturf(old_loc))
+		return
+	if(get_dist(src, old_loc) != 1) //teleporter shenanigans
+		return
+
+	var/mob/arriving_mob = arrived
+	var/old_loc_dir = get_dir(src, old_loc)
+	if(old_loc_dir & dir)
+		arriving_mob.client.move_delay += climb_slowdown //applied directly to client movedelay as mob delay would only apply after an additional movement
```

Comment:
Doesn't this cause slowdown when moving from one platform to another? Same with exit.

Reply:
? If two platforms were facing each other I guess. If they're facing the same way they're not going to proc.

Reply:
I was thinking about platforms as full tile things rather than edges.